libdnf/module/module_item.cpp

```diff
@@ -0,0 +1,276 @@
+    return modulemd_module_stream_get_context(md_stream);
+}
+
+
+std::string ModuleItem::get_context() const {
+    return libdnf::utils::string::c_to_str(modulemd_module_stream_get_context(md_stream));
+}
+
+
+const char * ModuleItem::get_arch_cstr() const {
+    return modulemd_module_stream_get_arch(md_stream);
+}
+
+
+std::string ModuleItem::get_arch() const {
+    return libdnf::utils::string::c_to_str(modulemd_module_stream_get_arch(md_stream));
+}
+
+
+std::string ModuleItem::get_name_stream(ModulemdModuleStream * md_stream) {
+    return libdnf::utils::sformat(
+        "{}:{}",
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_module_name(md_stream)),
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_stream_name(md_stream)));
+}
+
+
+std::string ModuleItem::get_name_stream_version() const {
+    return libdnf::utils::sformat(
+        "{}:{}:{}",
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_module_name(md_stream)),
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_stream_name(md_stream)),
+        std::to_string(modulemd_module_stream_get_version(md_stream)));
+}
+
+
+std::string ModuleItem::get_full_identifier() const {
+    return libdnf::utils::sformat(
+        "{}:{}:{}:{}:{}",
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_module_name(md_stream)),
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_stream_name(md_stream)),
+        std::to_string(modulemd_module_stream_get_version(md_stream)),
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_context(md_stream)),
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_arch(md_stream)));
+}
+
+
+std::string ModuleItem::get_summary() const {
+    return libdnf::utils::string::c_to_str(
+        modulemd_module_stream_v2_get_summary((ModulemdModuleStreamV2 *)md_stream, NULL));
+}
+
+
+std::string ModuleItem::get_description() const {
+    return libdnf::utils::string::c_to_str(
+        modulemd_module_stream_v2_get_description((ModulemdModuleStreamV2 *)md_stream, NULL));
+}
+
+
+std::vector<std::string> ModuleItem::get_artifacts() const {
+    std::vector<std::string> result_rpms;
+    char ** rpms = modulemd_module_stream_v2_get_rpm_artifacts_as_strv((ModulemdModuleStreamV2 *)md_stream);
+
+    for (char ** iter = rpms; iter && *iter; iter++) {
+        result_rpms.emplace_back(std::string(*iter));
+    }
+
+    g_strfreev(rpms);
+    return result_rpms;
+}
+
+
+std::vector<std::string> ModuleItem::get_demodularized_rpms() const {
+    std::vector<std::string> result_rpms;
+    char ** rpms = modulemd_module_stream_v2_get_demodularized_rpms((ModulemdModuleStreamV2 *)md_stream);
+
+    for (char ** iter = rpms; iter && *iter; iter++) {
+        result_rpms.emplace_back(std::string(*iter));
+    }
+
+    g_strfreev(rpms);
+    return result_rpms;
+}
+
+
+std::vector<ModuleProfile> ModuleItem::get_profiles(const std::string & name) const {
```

Comment:
What about to create a private method in ModuleItem `std::vector<ModuleProfile> get_profiles(const char * name)` and in .hpp ads inlines:
```

std::vector<ModuleProfile> ModuleItem::get_profiles(const std::string & name) const {
    return get_profiles(name.c_str());
}


std::vector<ModuleProfile> ModuleItem::get_profiles() const {
    return get_profiles(nullptr);
}
```

Reply:
That works, thanks. But when calling the `get_profiles(nullptr)` can't it call the `get_profiles(const std::string & name)` and then fail on `name.c_str()`?

Reply:
The private version can be named differently to prevent any confusion like `get_profiles_private` or `get_profiles_internal(const char * name)`.

Reply:
Yeah, I think it would be better. I renamed it.